### DISCUSSION TITLE
Store / Retrieve files GRPC

### DISCUSSION THREAD

**Discussion by william1benn** [seaweedfs/seaweedfs#5037] [Q&A]
seems there is not methods in proto files to retrieve or store files to master using grpc, are there any work arounds? Will this be something implemented in future?

**Comment by chrislusf** [comment]
what is the use case for it?

**Reply by william1benn** [reply]
my current use case, I am using seaweedfs in a swarm(microservice), with each system generating log files / metadata that are stored in weed in their respective collection. Sometimes we can say that these log files are being stored in parallel, we have not seen any performance downgrades from using json post request to store such data, yet the if we continue to add further systems the though currently is that we may need to run another master to handle the increase of requests. 

With this we do have plans to analyze this stored data from the collections, where the retrieval of files could benefit from the encoding of protobuf in terms of size.
